crates/bevy_ecs/src/system/builder.rs

```diff
@@ -545,11 +545,11 @@ mod tests {
             .build_state(&mut world)
             .build_system(
                 |mut p0: DynSystemParam, mut p1: DynSystemParam, mut p2: DynSystemParam| {
-                    let local = *p0.downcast_mut::<Local<usize>>().unwrap();
-                    let query_count = p1.downcast_mut::<Query<()>>().unwrap().iter().count();
-                    let _entities = p2.downcast_mut::<&Entities>().unwrap();
                     assert!(p0.downcast_mut::<Query<()>>().is_none());
-                    local + query_count
+                    let local: Local<usize> = p0.downcast_mut().unwrap();
+                    let query: Query<()> = p1.downcast_mut().unwrap();
+                    let _entities: &Entities = p2.downcast_mut().unwrap();
+                    *local + query.iter().count()
```

Comment:
I understand that this change you're proposing allows for the variable type to infer the type parameter on `downcast/_mut`, but this change feels rather pointless. In the case of `query_count`, I'd even say it's a regression to move away from turbofish syntax.

Reply:
Yeah, I had this here to check that inference actually worked, but @alice-i-cecile's suggestion to make a dedicated test for that means I can put this back.